src/coreclr/jit/gentree.h

```diff
@@ -7154,6 +7156,39 @@ inline GenTree* GenTree::gtEffectiveVal(bool commaOnly /* = false */)
     }
 }
 
+//-------------------------------------------------------------------------
+// gtUnwrapCSE - find value being CSE'd
+//
+// Returns:
+//    tree representing value being CSE'd, if this tree represents a
+//    comma-wrapped CSE definition and use.
+//
+//    original tree, of not.
+//
+inline GenTree* GenTree::gtUnwrapCSE()
+{
```

Comment:
While this is the format that we use for a CSE def there are several other places we will generated this patterm.
Looking through the code base this function also is commonly used to generate an assignment to a new temp LclVar


`GenTree* Compiler::fgInsertCommaFormTemp(GenTree** ppTree, CORINFO_CLASS_HANDLE structType /*= nullptr*/)`

Reply:
It doesn't have to be a CSE, it could be any value. I can rename the helper.

During global prop we have a mixture of rules that key off VNs and also look at IR shapes. In general we should just be able to use the VNs to figure most things out, but that doesn't happen consistently today. So in this case I need to find the original IR shape.